src/zone/scheduler.h

```diff
@@ -48,6 +48,15 @@ namespace zone {
         /// </remarks>
         void ScheduleOnWorker(WorkerId workerId, std::shared_ptr<Task> task);
 
+        /// <summary> Schedules immediate task on a specific worker. </summary>
+        /// <param name="workerId"> The id of the worker. </param>
+        /// <param name="task"> Task to schedule. </param>
+        /// <remarks>
+        /// By design, it enqueues a task immediately before any non-immedidate task,
+        /// but after any existing immediate task.
+        /// </remarks>
+        void ScheduleImmediateOnWorker(WorkerId workerId, std::shared_ptr<Task> task);
+
```

Comment:
Create a enum type `TaskType` and add as function `ScheduleOnWorker`'s second parameter, instead of creating a new method

Reply:
Fixed by adding SchedulePhase as discussed yesterday.